.github/workflows/main.yml

```diff
@@ -93,24 +103,34 @@ jobs:
     needs: [test]
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: "3.12"
+          version: "0.4.12"
+          enable-cache: true
+
+      - name: Set up Python 3.12
+        run: uv python install 3.12
+
+      - name: Create venv
+        run: uv venv --python 3.12
+
       - name: get coverage files
         uses: actions/download-artifact@v4
         with:
           merge-multiple: true
           path: coverage
-      - run: pip install coverage[toml]
-      - run: ls -lh coverage
-      - run: coverage combine coverage
-      - run: coverage xml
+
+      - run: uv pip install coverage[toml]
+      - run: uv run coverage combine coverage
+      - run: uv run coverage xml
```

Comment:
Is there not a way to 'activate' uv so that we don't have to remember to always use `uv run`? What about scripts that internally just use `python` or whatever other command?

Reply:
Apparently the modern take is that activating environments is an anti-pattern we should now avoid.

I suspect you can active `uv`, but commands like this should not assume you have.